### PR TITLE
[Drop-In UI] add option to specify custom location puck with drop-in

### PR DESCRIPTION
**Pull request by Zayankovsky**

### CODE REVIEW

**Review by tomaszrybakiewicz — APPROVED** [review]
LGTM
@Zayankovsky Can you add description to this PR?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Mapbox welcomes participation and contributions from everyone.
 - Moved `MapboxAudioGuidance` and `MapboxAudioGuidanceState` into public api. [#6336](https://github.com/mapbox/mapbox-navigation-android/pull/6336)
 - Introduced `ViewOptionsCustomization.showCameraDebugInfo` to allow end users to enable camera debug info. [#6356](https://github.com/mapbox/mapbox-navigation-android/pull/6356)
 - Added `ComponentInstaller` for the `LocationComponent` that offers simplified integration of map LocationPuck. [#6206](https://github.com/mapbox/mapbox-navigation-android/pull/6206)
+- Added `ViewStyleCustomization.locationPuck` that allows to define a custom location puck for `MapView`. [#6365](https://github.com/mapbox/mapbox-navigation-android/pull/6365)
```
Comment (abhishek1508) [inline CHANGELOG.md:10]:
```suggestion
- Added `ViewStyleCustomization.locationPuck` that allows to define a custom location puck using `NavigationView`. [#6365](https://github.com/mapbox/mapbox-navigation-android/pull/6365)
- Replaced `ViewStyleCustomization.defaultMarkerAnnotationOptions` with `ViewStyleCustomization.defaultDestinationMarkerAnnotationOptions`
```

File: libnavigation-core/api/current.txt [hunk 2]
```diff
@@ -676,11 +676,13 @@ package com.mapbox.navigation.core.routealternatives {
   }
 
   public final class AlternativeRouteMetadata {
+    method public int getAlternativeId();
```
Comment (abhishek1508) [inline libnavigation-core/api/current.txt:679]: Did we miss these docs in some other PR?

**Review by abhishek1508 — COMMENTED** [review]
@Zayankovsky

**Comment by tomaszrybakiewicz** [comment]
@abhishek1508 
That stutter is definitely on the Maps side. Puck position is animated like this:



I can only assume that when the map style changes mid-animation, the animated style layer is destroyed along with its animator. The remaining animation frames are "lost". 
When the new style layer is installed, the new animation starts on next `LocationComponentPluginImpl.updateCurrentPosition(...)` call (which is every second), hence the stutter.

File: libnavigation-core/api/current.txt (hunk 2 above)
Comment (Zayankovsky) [reply libnavigation-core/api/current.txt:679]: Yes, @VysotskiVadim forgot to update these files in #6329.

**Review by abhishek1508 — APPROVED** [review]
Thanks for addressing the feedback!

**Comment by Zayankovsky** [comment]
> @Zayankovsky I am seeing a stutter when changing the puck at runtime while in active navigation. 

It should be better now, looks like it was caused by repeatedly setting the same location provider, so I added a check to avoid this. Also I removed a bunch of code from `LocationPuckComponent`, we don't need to wait for location updates or for style to load before updating location plugin.

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/LocationPuckComponent.kt [hunk 3]
```diff
@@ -25,49 +17,12 @@ class LocationPuckComponent(
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
 
-        coroutineScope.launch {
-            locationProvider.waitForFirstLocation()
-            map.getStyle {
```
Comment (dzinad) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/LocationPuckComponent.kt:30]: Why is it not needed anymore? And waiting for the first location.

Comment (Zayankovsky) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/LocationPuckComponent.kt:30]: Waiting for the first location was added in #5555. I think it was an issue with `getStyle`, which is now also removed. We don't do any of that in 1TAP and never had any issues with location puck.

**Pull request merged**